crates/superposition_types/src/lib.rs

```diff
@@ -487,3 +491,27 @@ pub struct PaginatedResponse<T> {
     pub total_items: i64,
     pub data: Vec<T>,
 }
+
+/// Provides struct to directly extract query params as serde_json::Map<String, serde_json::Value>
+#[derive(Deserialize)]
+#[serde(from = "HashMap<String,String>")]
+pub struct QueryMap(Map<String, Value>);
+
+impl QueryMap {
+    pub fn into_inner(self) -> Map<String, Value> {
+        self.0
+    }
+}
+
+impl From<HashMap<String, String>> for QueryMap {
+    fn from(value: HashMap<String, String>) -> Self {
+        let value = value
+            .into_iter()
+            .map(|(key, value)| {
+                (key, value.parse().unwrap_or_else(|_| Value::String(value)))
+            })
+            .collect::<Map<_, _>>();
+
+        Self(value)
+    }
+}
```

Comment:
there are no errors here to be handled

Reply:
@ayushjain17 When you do value.parse(), it returns an error if it cannot parse the config right? The unwrap_or_else function that forces it into Value::String is not correct, since if it were a valid string it wouldn't have thrown an error during parse.

Reply:
no, this is not how this parse works, this is parse works exactly in the same way JSON.parse works in javascript, `"{'a':'b'}"`, is a stringified object which gets parsed here to `{'a' : 'b'}`
similarly, `"123"` is a stringified int, gets parsed to `123`
`"text"` is a string, which fails in parse, and in that case you where wanting to get a string itself `"text"`

Reply:
also, I am just refactoring this thing, this was already implemented as an [extractor function](https://github.com/juspay/superposition/pull/264/files#diff-8eeff1524d054e7ce25f18154a5f85046532f355a95bdc550988bc7470e7fb4aL123)